packages/dds/sequence/src/test/intervalCollection.snapshot.spec.ts

```diff
@@ -0,0 +1,148 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import { ReferenceType } from "@fluidframework/merge-tree";
+import {
+    MockFluidDataStoreRuntime,
+    MockContainerRuntimeFactory,
+    MockStorage,
+} from "@fluidframework/test-runtime-utils";
+import { ISummaryTree } from "@fluidframework/protocol-definitions";
+import { SharedString } from "../sharedString";
+import { SharedStringFactory } from "../sequenceFactory";
+import { IntervalCollection, IntervalType, SequenceInterval } from "../intervalCollection";
+
+const assertIntervals = (
+    sharedString: SharedString,
+    intervalCollection: IntervalCollection<SequenceInterval>,
+    expected: readonly { start: number; end: number; }[],
+    validateOverlapping: boolean = true,
+) => {
+    const actual = Array.from(intervalCollection);
+    if (validateOverlapping && sharedString.getLength() > 0) {
+        const overlapping = intervalCollection.findOverlappingIntervals(0, sharedString.getLength() - 1);
+        assert.deepEqual(actual, overlapping, "Interval search returned inconsistent results");
+    }
+    assert.strictEqual(actual.length, expected.length,
+        `findOverlappingIntervals() must return the expected number of intervals`);
+
+    const actualPos = actual.map((interval) => {
+        assert(interval);
+        const start = sharedString.localRefToPos(interval.start);
+        const end = sharedString.localRefToPos(interval.end);
+        return { start, end };
+    });
+    assert.deepEqual(actualPos, expected, "intervals are not as expected");
+};
+
+async function loadSharedString(
+    containerRuntimeFactory: MockContainerRuntimeFactory,
+    id: string,
+    summary: ISummaryTree,
+): Promise<SharedString> {
+    const dataStoreRuntime = new MockFluidDataStoreRuntime();
+    const containerRuntime = containerRuntimeFactory.createContainerRuntime(dataStoreRuntime);
+    (containerRuntime as any).deltaManager.lastSequenceNumber = containerRuntimeFactory.sequenceNumber;
```

Comment:
this is pretty ugly. Is there a better way to do this? Context is that after a shared-string is loaded for collaboration in this fashion, it won't necessarily have its sequence number updated to match that of the `containerRuntimeFactory`. That becomes problematic when trying to resolve submitted ops: without this line, the first operation will think it's seq 1, even though it should be later (seq 1 should be the "ABCDEF" insert at the start of the document, conceptually).

Seems like somewhat of a shortcoming in mocks, but I'm always wary of trying to over-complicate mock fidelity. One way or another though we do need some test coverage for DDSes of "loading from a summary then collaborating more," since summarization is tricky from that perspective. Maybe just moving this to e2e would be better. Open to suggestions, though this fix is somewhat time-sensitive :)

Reply:
you need to access it after creating the container runtime, as it will be overwritten